File: src/main/kotlin/Java/Hotel/SistemaReservas.java
package Java.Hotel;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

class SistemaReservas {
    public List<Reserva> reservas;

    public SistemaReservas() {
        reservas = new ArrayList<>();
    }

    public void reservarHabitacion(Habitacion habitacion, Cliente cliente, Date fechaInicio, Date fechaFin) {
        Reserva nuevaReserva = new Reserva(habitacion, cliente, fechaInicio, fechaFin);
        reservas.add(nuevaReserva);
        System.out.println("Reserva realizada: " + nuevaReserva);
    }

    public void cancelarReserva(Reserva reserva) {
        reservas.remove(reserva);
        System.out.println("Reserva cancelada: " + reserva);
    }

    public void mostrarReservasActivas() {
        System.out.println("Reservas activas:");
        for (Reserva reserva : reservas) {
            System.out.println(reserva);
        }
    }
}

File: src/main/kotlin/kotlinOPP/Hotel/SistemaReservas.kt
package kotlinOPP.Hotel

import java.util.*

class SistemaReservas {
    val reservas: MutableList<Reserva> = mutableListOf()

    fun reservarHabitacion(habitacion: Habitacion, cliente: Cliente, fechaInicio: Date, fechaFin: Date) {
        val reserva = Reserva(habitacion, cliente, fechaInicio, fechaFin)
        reservas.add(reserva)
        println("Reserva realizada: $reserva")
    }

    fun cancelarReserva(reserva: Reserva) {
        if (reservas.remove(reserva)) {
            println("Reserva cancelada: $reserva")
        } else {
            println("No se encontró la reserva para cancelar.")
        }
    }

    fun mostrarReservasActivas() {
        if (reservas.isEmpty()) {
            println("No hay reservas activas.")
        } else {
            println("Reservas activas:")
            reservas.forEach { println(it) }
        }
    }
}